flink-tests/src/test/java/org/apache/flink/test/checkpointing/SavepointITCase.java

```diff
@@ -515,6 +517,50 @@ public void testTriggerSavepointWithCheckpointingDisabled() throws Exception {
         }
     }
 
+    @Test
+    public void testTriggerSavepointWithoutCheckpointBaseLocations() throws Exception {
+        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+        env.getCheckpointConfig().disableCheckpointing();
+        env.setParallelism(1);
+
+        env.addSource(new IntegerStreamSource()).addSink(new DiscardingSink<>());
+
+        JobGraph jobGraph = env.getStreamGraph().getJobGraph();
+
+        Configuration config = getFileBasedCheckpointsConfig();
+        config.addAll(jobGraph.getJobConfiguration());
+
+        MiniClusterWithClientResource cluster =
+                new MiniClusterWithClientResource(
+                        new MiniClusterResourceConfiguration.Builder()
+                                .setConfiguration(config)
+                                .setNumberTaskManagers(1)
+                                .setNumberSlotsPerTaskManager(1)
+                                .build());
+        cluster.before();
+        ClusterClient<?> client = cluster.getClusterClient();
+
+        String savepointPath = null;
+        try {
+            client.submitJob(jobGraph).get();
+
+            waitForAllTaskRunning(cluster.getMiniCluster(), jobGraph.getJobID(), false);
+
+            savepointPath = client.triggerSavepoint(jobGraph.getJobID(), null).get();
+
+            assertNotNull(savepointPath);
+
+            client.cancel(jobGraph.getJobID()).get();
+            // checkpoint directory should not be initialized
+            assertEquals(0, Objects.requireNonNull(checkpointDir.listFiles()).length);
```

Comment:
Why do you use `listFiles` instead of `exists` here?

Reply:
The children files of `checkpointDir` should be empty if the checkpoint based location is not initialized. Of course we can use `exists` with `checkpointDir/{jobID}` to achieve the same goal.